plans/dht/README.md

```diff
@@ -0,0 +1,70 @@
+# `Plan:` The go-libp2p DHT behaves
+
+IPFS can safely rely on the latest DHT upgrades by running go-libp2p DHT tests directly
+
+## What is being optimized (min/max, reach)
+
+- (Minimize) Numbers of peers dialed to as part of the call
+- (Minimize) Number of failed dials to peers
+- (Minimize) Time that it took for the test
+- (Minimize) Routing efficiency (# of hops (XOR metric should decrease at every step))
+
+## Plan Parameters
+
+- **Network Parameters**
+  - `Region` - Region or Regions where the test should be run at (default to single region)
+  - `N` - Number of nodes that are spawn for the test (from 10 to 1000000)
+- **Image Parameters**
+  - Single Image - The go-libp2p commit that is being tested
+  - Image Resources CPU & Ram
+
+## Tests
+
+### `Test:` Find Peers
+
+- **Test Parameters**
+  - `random-walk` - Automatic random-walk On/Off
+  - `bucket-size` - Kademlia DHT bucket size  
+  - `n-find-peers` - Number of times a Find Peers call is executed from each node (picking another node PeerId at random)
+- **Narrative**
+  - **Warm up**
+    - All nodes boot up
+    - Each node as it boots up, connects to the node that previously joined

```

Comment:
Is the daisy chaining by design? Could be a pretty unrealistic scenario if random walks are turned off?

Reply:
Yes, the _ring connectivity_ is the minimum and simplest topology that enables full network connectivity for any one. If random-walks are enabled, they will quickly update their routing tables, if they are disabled, then it will be the FindPeer call that will also shape each routing table for the first time.